lang/interpolate/parse.rl

```diff
@@ -76,7 +76,7 @@ func Parse(data string) (out Stream, _ error) {
 		var = '${' var_name '}' ;
 
 		# Any special escape characters are matched here.
-		escaped_lit = '\\' ( 'a' | 'b' | 'f' | 'n' | 'r' | 't' | 'v' | '\\' | '"' | '$' )
+		escaped_lit = '\\' (any)
```

Comment:
don't we want to _not_ allow the non-registered escape characters?

Reply:
and as it turns out a previous test i wrote caught this change and so this PR failed.

Reply:
My justification is that allowing any char in here gives a better error message instead of a generic "sorry no can do". It's also a bit redundant to check the character twice.

Reply:
Updated both the tests and the error message to be a little more human friendly